src/http-api/resources/files.js

```diff
@@ -0,0 +1,67 @@
+'use strict'
+
+const bs58 = require('bs58')
+const ndjson = require('ndjson')
+const multipart = require('ipfs-multipart')
+const debug = require('debug')
+const log = debug('http-api:files')
+log.error = debug('http-api:files:error')
+
+exports = module.exports
+
+exports.add = {
+  handler: (request, reply) => {
+    if (!request.payload) {
+      return reply('Array, Buffer, or String is required').code(400).takeover()
+    }
+
+    const parser = multipart.reqParser(request.payload)
+
+    var file = false
+    var written = 0
+
+    var serialize = ndjson.serialize()
+    // hapi doesn't permit object streams: http://hapijs.com/api#replyerr-result
+    serialize._readableState.objectMode = false
+
+    var fileAdder = request.server.app.ipfs.files.add()
+
+    fileAdder.on('data', (file) => {
+      serialize.write({
+        Name: file.path,
+        Hash: bs58.encode(file.multihash).toString()
```

Comment:
Please use `mh.toB58String(file.multihash)` from the `multihashes` package insteada